src/NuGet.Core/NuGet.Packaging/NuspecReader.cs

```diff
@@ -471,7 +471,29 @@ public LicenseMetadata GetLicenseMetadata()
                             {
                                 try
                                 {
+                                    IList<string> invalidLicenseIdentifiers = null;
                                     var expression = NuGetLicenseExpression.Parse(license);
+                                    Action<NuGetLicense> licenseProcessor = delegate (NuGetLicense nugetLicense)
+                                    {
+                                        if (!nugetLicense.IsStandardLicense)
+                                        {
+                                            if (invalidLicenseIdentifiers == null)
+                                            {
+                                                invalidLicenseIdentifiers = new List<string>();
+                                            }
+                                            invalidLicenseIdentifiers.Add(nugetLicense.Identifier);
+                                        }
+                                    };
+                                    expression.OnEachLeafNode(licenseProcessor, null);
```

Comment:
its so hard to read what exactly you're trying to do here...

Reply:
I will add some comments. 
But basically this is a visitor for each leaf node.